articles/tokens/concepts/token-storage.md

```diff
@@ -0,0 +1,104 @@
+- [Regular Web App Quickstarts](/quickstart/webapp)
+
+## Native/Mobile app scenarios
+
+Store tokens in a secure storage that the OS offers and limit access to that storage. For example, leverage KeyStore for Android and KeyChain for iOS.
+
+Use the following flow types in these scenarios:
+
+- [Authorization Code Flow with Proof Key for Code Exchange](/flows/concepts/auth-code-pkce)
+- [Saving and Renewing Tokens for Android](/libraries/auth0-android/save-and-refresh-tokens)
+- [Saving and Renewing Tokens for Swift](/libraries/auth0-swift/save-and-refresh-jwt-tokens)
+- [Native/Mobile Apps Quickstarts](/quickstart/native)
+
+## Single-page app scenarios
+
+We recommend using the [Auth0 Single Page App SDK](/libraries/auth0-spa-js). The Auth0 SPA SDK handles token storage, session management, and other details for you.
+
+When the SPA calls only an API that is served from a domain that can share cookies with the domain of the SPA, no tokens are needed. OAuth adds additional attack vectors without providing any additional value and should be avoided in favor of a traditional cookie-based approach. For an overview of this approach and an example implementation, see [Single-Page App Authentication Using Cookies](/login/spa/authenticate-with-cookies).
+
+When the SPA calls multiple APIs that reside in a different domain, access and optionally refresh tokens are needed.
+
+-  If the SPA backend can handle the API calls, handle tokens server-side using:
+    - [Authorization Code Flow](/flows/concepts/auth-code)
+    - [Authorization Code Flow with Proof Key for Code Exchange (PKCE)](/flows/concepts/auth-code-pkce)
+    - [Hybrid Flow](/api-auth/grant/hybrid)
+
+- If the SPA backend cannot handle the API calls, the tokens should be stored in the SPA backend but the SPA needs to fetch the tokens from the backend to perform requests to the API. A protocol needs to be established between the backend and the SPA to allow the secure transfer of the token from the backend to the SPA.
+
+- If you have a SPA with **no** corresponding backend server, your SPA should request new tokens on login and store them in memory without any persistence. To make API calls, your SPA would then use the in-memory copy of the token.
+
+### Browser in-memory scenarios
+
+Auth0 recommends storing tokens in browser memory as the most secure option. Using [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) to handle the transmission and storage of tokens is the best way to protect the tokens, as Web Workers run in a separate global scope than the rest of the application. Use [auth0-spa-js](https://github.com/auth0/auth0-spa-js) whose default storage option is in-memory storage leveraging Web Workers.
+
+If you cannot use Web Workers, Auth0 recommends as an alternative that you use [JavaScript closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#Emulating_private_methods_with_closures) to emulate private methods.
```

Comment:
```suggestion
If you cannot use Web Workers, Auth0 recommends as an alternative that you use [JavaScript closures](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#Emulating_private_methods_with_closures) to emulate private methods.

 Use [auth0-spa-js](https://github.com/auth0/auth0-spa-js) whose default storage option is in-memory storage. It leverages both Web Workers and JavaScript closures depending on the type of token
```

Reply:
> whose default storage option is in-memory storage leveraging Web Workers.

Just a question - the SPA SDK will leverage web workers where possible, except for IE11 where they are supported, but have other complications. Do we want to make reference to anything about that here, to the effect of "It leverages both Web Workers (in modern browsers), and..."

We could link to some readme docs that explains web workers and browser support specific to the SDK, in more detail.

Reply:
It's not only IE11. Access Tokens can't be stored in Web Workers. This document is about tokens in general.
Linking to some readme docs is a good idea!

Reply:
The document is, but this paragraph in particular is in the context of the SPA SDK. Perhaps it's not the right place for that kind of detail, but let's link to specifics.

Reply:
Right and in the SPA SDK we use Web Workers for RT in modern browsers and JavaScript closures for ATs and for RTs in IE11. Am I missing something?

Reply:
No you're correct, but the paragraph doesn't mention modern browsers - my question (and it was only a question) was whether or not it should.